### PR TITLE
Fail if conda-forge.yml says azure, but still uses appveyor

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge/core, this will fail PRs needing a rerender after the appveyor to azure change. This will not fail on master, only PRs.

**Comment by isuruf** [comment]
ping

**Comment by jakirkham** [comment]
Sometimes the latest re-render has problems or a feedstock needs to manually fix the CI scripts temporarily. How does this handle those cases?

File: recipe/conda_forge_ci_setup/build_utils.py [hunk 1]
```diff
@@ -42,26 +43,62 @@ def update_global_config(feedstock_root):
             _global_config[k1][k2] = repo_config[k1][k2]
 
 
+def fail_if_outdated_windows_ci(feedstock_root):
+    if sys.platform != "win32":
+        return
+
+    if "APPVEYOR_ACCOUNT_NAME" in os.environ:
+        provider = "appveyor"
+        if os.environ["APPVEYOR_ACCOUNT_NAME"] != "conda-forge":
```
Comment (scopatz) [inline recipe/conda_forge_ci_setup/build_utils.py]: This is sort of a weird condition. What is it protecting against?

**Comment by scopatz** [comment]
Does this prevent feedstocks from using appveyor as a provider at all?

**Comment by isuruf** [comment]
What this prevents is when you have `provider: win: azure` in `conda-forge.yml` and a `.appveyor.yml` file in the repo. Example given  by @scopatz and @jakirkham won't get affected. Have a look at the code.

File: recipe/conda_forge_ci_setup/build_utils.py (hunk 1 above)
Comment (isuruf) [reply recipe/conda_forge_ci_setup/build_utils.py]: > What is it protecting against?

Breaking other organizations using conda-smithy.

**Comment by scopatz** [comment]
Ahh that makes more sense.

**Comment by jakirkham** [comment]
Yeah the title and description of this PR were a bit vague. Maybe they should be updated to reflect more clearly what is being proposed here.

**Comment by isuruf** [comment]
Fixed the title

**Comment by scopatz** [comment]
this LGTM.  @jakirkham?

**Comment by jakirkham** [comment]
I'll defer to you.

**Comment by isuruf** [comment]
ping

**Pull request merged**